### DISCUSSION TITLE
SQL (Many) Relationship not updated on changes

### DISCUSSION THREAD

**Discussion by gerhardmpl** [Budibase/budibase#4590] [Help]
Hi everybody, trying out a `Many` relationship scenario with a MySQL Datasource and three tables: `students`, `courses` and a _Through_ `student_course` table, where the `student_course` table acts as a junction table.

- students (student_id (PK), first_name, last_name)
- courses (course_id (PK), course_title)
- student_course (student_id (PK), course_id (PK))

Each student can take multiple courses. 
Each course can have multiple students.

Setting this up in Budibase [like described in the documentation](https://docs.budibase.com/building-apps/data/tables/postgresql-mysql-and-sql-server) and adding courses for a student (or the other way around) works.
But if you deselect a previously selected course for a student, the junction table is not updated, and the relationship stays.
When using the internal Budibase DB in a seperate scenario/app, the relationship is updated in both directions as expected.
Attached please find the sql file to import the tables to an _existing_ database.

Not sure if this is a bug or if it works as designed (in which case I would appreciate any help or direction on how to update the SQL (Many) relationship)

[students_courses.sql.txt](https://github.com/Budibase/budibase/files/8104932/students_courses.sql.txt)

**Comment by mjashanks** [comment]
@gerhardmpl it took a while, but I think I have just figured out the issue. This is a bug. However, there is a woraround - if you're willing to use it!

It seems like Budibase does not like compsite primary keys. If you add a new `id` column, and make it the primary key - then re-fetch tables, I believe that this will work for you.

I will log this as a bug.

**Reply by mjashanks** [reply]
logged here: https://github.com/Budibase/budibase/issues/4604

**Reply by mjashanks** [reply]
Also - someone has had this issue recently - and it seems to have nothing to do with what I suggested - maybe there's more to it.... https://github.com/Budibase/budibase/discussions/4408
